R/render.r

```diff
@@ -422,7 +422,121 @@ data_deps <- function(pkg, depth) {
       )
     )
   )
-  deps <- bslib::bs_theme_dependencies(bs_theme)
+
+  # map either secondary OR fg, bg to component-active-bg
+  # and also dropdown-link-active-bg
+  # unless a value was set by the user
+  if (!is.null(pkg$meta[["template"]]$bslib$secondary)) {
+    if (is.null(pkg$meta[["template"]]$bslib$`component-active-bg`)) {
+      bs_theme <- bslib::bs_add_variables(
+        bs_theme,
+        "component-active-bg" = as.character(
+          bslib::bs_get_variables(bs_theme, "secondary")
+        )
```

Comment:
What's the rationale behind this? It seems like it could be quite surprising if `$primary` doesn't impact `$component-active-bg`

Reply:
Ok, to be discussed with @apreshill

Reply:
Hi @cpsievert- we went back and forth on this a lot, and really appreciate your input as we are both newer to Bootstrap than you are. Part of our rationale was that we wanted to keep the navbar and sidebar TOC active background colors the same, and having `$primary` cascade as the active background color to both of those elements felt a little aggressive (especially as it needs to be dark enough to get enough color contrast with text for link colors to meet accessibility guidelines).

The pkgdown dev site (https://pkgdown.r-lib.org/dev/articles/pkgdown.html) and `bookdown::bs4_book` follow this pattern, where the active background color is more of a shaded accent color and not suitable for text but good as a bg color.

We mapped those specific elements to `$component-active-bg` but perhaps that was the wrong BS4 variable to use? What variables do you think would be least surprising to map onto these elements?

<img width="1333" alt="Screen Shot 2021-07-13 at 3 50 07 PM" src="https://user-images.githubusercontent.com/12160301/125535599-e5ab1da8-53e5-44fd-8550-b59b587b9e6b.png">

<img width="1333" alt="Screen Shot 2021-07-13 at 3 57 22 PM" src="https://user-images.githubusercontent.com/12160301/125535886-83612804-4b90-45ff-8f74-054ac0886568.png">

Reply:
Ah ok, thanks for the detailed explanation. Is there any reason why those styles couldn't use `$secondary` directly instead of `$component-active-bg`? Also it might be worth having the foreground active styles default to `color-contrast($secondary)` so that one could set a dark bg color and have the contrasting 'just work'

Reply:
Actually thinking about this a bit more it might make a bit more sense to use [`$list-group-active-bg`](https://rstudio.github.io/bslib/articles/bs5-variables.html#list-group-active-bg) since semantically list groups are very close to TOC

Reply:
Thank you! I would not have thought of that variable. So `$secondary` would affect:

+ `$nav-pills-link-active-bg`
+ `$dropdown-link-active-bg`
+ `$list-group-active-bg` (this would be the active content section in the sidebar TOC)
+ plus the code copy button hover/active background color

The foreground active styles would then be:
+ `$nav-pills-link-active-color`
+ `$dropdown-link-active-color`
+ `$list-group-active-color` (this would be the active content section in the sidebar TOC)
+ plus the active copy button icon color

Setting the default for all four of those to be `color-contrast($secondary)` makes a lot of sense, but I would also want to offer a single variable that would help users customize that quickly too. Previously, we were using `$primary` there- do you have an idea for a Bootstrap variable that might help us tie these elements together?

Reply:
Summing up discussion with @cpsievert here (pls correct me if I'm wrong!). 

Some of these elements we'd like for users to be able to customize simply may not have existing Bootstrap variables that make sense. 

For example, `$nav-pills-link-active-bg` may not make sense to apply to the navbar, b/c a user could just want to add this as an element to a page outside of the navbar and then they could not make it look different than the navbar (which has a background color). In this case, we may want to create a new but non-Bootstrap variable in the `pkgdown.scss` file at the top to define it with a Bootstrap variable as a default like:

```
# new variable: default is a bootstrap variable
$navbar-active-bg: $secondary !default;
$navbar-active-color: color-contrast($secondary);
```

The main challenge then is that a user can only know the variable name here with our docs- you wouldn't be able to use Bootstrap to figure out how to customize.

A "rule" we could adopt is: 

> The fg color will always be set after the bg color, and defaults to a color-contrast of the bg color. 

If we *also* create a custom variable for the fg color (like above, with `$navbar-active-color`), users could still override that default variable (again, assuming they have access to docs and aren't trying to use Bootstrap to figure out the right variable name).

On the other hand, `$list-group-active-bg` would make more sense for the sidebar TOC.

TODO: @apreshill will make a list of the variables we'd like to use from Bootstrap, and the missing ones we need to add.

Reply:
Another advantage of defining pkgdown's own variables is that the logic is simpler: they are either set by pkgdown with `!default` or set by the user, whereas currently there are cases where we want to differentiate between set by BS / set by pkgdown / set by the user (the cases where if a variable is not set by the user, we have some sort of new variable inheritance).